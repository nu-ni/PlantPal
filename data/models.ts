export type PlantCollection = {
    id?: number;
    title: string;
    lastActive: Date;
    count?: number;
};

export type Plant = {
    id?: number;
    title: string;
    frequency: number;
    waterAmount: number;
    collectionId: number;
    image: string;
};

export type Dict = {
    [key: string]: string;
};
