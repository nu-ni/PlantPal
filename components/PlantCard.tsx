import { Pressable, StyleSheet, Text, Image } from 'react-native';
import { View } from 'react-native';
import React from 'react';
import { Plant } from '../data/models';

type CardProps = {
  item: Plant;
  onItemSelected?: () => void
};

export default function PlantCard({ item, onItemSelected }: CardProps) {
  const imageSource = { uri: `data:image/jpeg;base64,${item.image}` };
  
  return (
    <Pressable onPress={onItemSelected}>
      <View style={styles.card}>
        <Image source={imageSource} style={styles.image}/>
        <View style={styles.titleContainer}>
          <Text style={styles.title}>{item.title}</Text>
          <Text style={styles.title}>{item.frequency + ' d'}</Text>
          <Text style={styles.title}>{item.waterAmount + ' ml'}</Text>
        </View>
      </View>
    </Pressable>
  );
}

const styles = StyleSheet.create({
  image: {
    height: '100%',
    width: '50%',
    alignSelf: 'center',
  },
  card: {
    backgroundColor: 'white',
    borderRadius: 5,
    flexDirection: 'row',
    overflow: 'visible',
    padding: 10,
    marginBottom: 10,
  },
  titleContainer: {
    flex: 9,
    justifyContent: 'center',
    marginVertical: 17,
  },
  title: {
    marginRight: 13,
    textAlign: 'right',
    fontSize: 20,
  },
});