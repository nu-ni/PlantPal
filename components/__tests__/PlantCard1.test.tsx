// failt leider

import React from 'react';
import PlantCard from '../PlantCard';
import { render, fireEvent } from '@testing-library/react-native';
import renderer from 'react-test-renderer';
import { Plant } from '@/data/models';


// Prepare dummy data
const plantDummy: Plant = {
    id: 1,
    title: 'Plant 1',
    frequency: 3,
    waterAmount: 200,
    image: 'base64image',
    collectionId: 0
};

describe('<PlantCard />', () => {
  // Setup
  const onItemSelectedMock = jest.fn();
  const card = <PlantCard item={plantDummy} onItemSelected={onItemSelectedMock} />;

  // Snapshot test
  it('renders correctly and matches the snapshot', () => {
    const tree = renderer.create(card).toJSON();
    expect(tree).toMatchSnapshot();
  });

  // Tests
  it('renders correctly with the correct item details displayed', () => {
    const { getByText } = render(card);

    // Check title
    const titleElement = getByText(plantDummy.title);
    expect(titleElement).toBeTruthy();
    expect(titleElement.props.children).toEqual(plantDummy.title);

    // Check frequency
    const frequencyElement = getByText(`${plantDummy.frequency} d`);
    expect(frequencyElement).toBeTruthy();
    expect(frequencyElement.props.children).toEqual(`${plantDummy.frequency} d`);

    // Check water amount
    const waterAmountElement = getByText(`${plantDummy.waterAmount} ml`);
    expect(waterAmountElement).toBeTruthy();
    expect(waterAmountElement.props.children).toEqual(`${plantDummy.waterAmount} ml`);
  });

  it('calls onItemSelected when card is pressed', () => {
    const { getByText } = render(card);
    const cardComponent = getByText(plantDummy.title);

    fireEvent.press(cardComponent);
    expect(onItemSelectedMock).toHaveBeenCalledTimes(1);
  });
});
