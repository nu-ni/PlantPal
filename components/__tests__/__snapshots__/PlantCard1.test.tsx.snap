// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PlantCard /> renders correctly and matches the snapshot 1`] = `
<View
  accessibilityState={
    {
      "busy": undefined,
      "checked": undefined,
      "disabled": undefined,
      "expanded": undefined,
      "selected": undefined,
    }
  }
  accessibilityValue={
    {
      "max": undefined,
      "min": undefined,
      "now": undefined,
      "text": undefined,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={true}
  onBlur={[Function]}
  onClick={[Function]}
  onFocus={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
>
  <View
    style={
      {
        "backgroundColor": "white",
        "borderRadius": 5,
        "elevation": 5,
        "flexDirection": "row",
        "marginBottom": 10,
        "overflow": "visible",
        "padding": 10,
        "shadowColor": "#000",
        "shadowOffset": {
          "height": 2,
          "width": 0,
        },
        "shadowOpacity": 0.45,
        "shadowRadius": 4.84,
      }
    }
  >
    <Image
      source={
        {
          "uri": "data:image/jpeg;base64,base64image",
        }
      }
      style={
        {
          "alignSelf": "center",
          "height": "100%",
          "width": "50%",
        }
      }
    />
    <View
      style={
        {
          "flex": 9,
          "justifyContent": "center",
          "marginVertical": 17,
        }
      }
    >
      <Text
        style={
          {
            "fontSize": 20,
            "marginRight": 13,
            "textAlign": "right",
          }
        }
      >
        Plant 1
      </Text>
      <Text
        style={
          {
            "fontSize": 20,
            "marginRight": 13,
            "textAlign": "right",
          }
        }
      >
        3 d
      </Text>
      <Text
        style={
          {
            "fontSize": 20,
            "marginRight": 13,
            "textAlign": "right",
          }
        }
      >
        200 ml
      </Text>
    </View>
  </View>
</View>
`;
